parcels/fieldfilebuffer.py

```diff
@@ -188,36 +188,47 @@ def __init__(self, *args, **kwargs):
+                         'lon': ['lon', 'nav_lon', 'x', 'longitude', 'lo', 'ln', 'i', 'XC', 'XG']}
     _min_dim_chunksize = 16
 
     """ Class that encapsulates and manages deferred access to file data. """
     def __init__(self, *args, **kwargs):
+        """
+        Initializes this specific filebuffer type. As a result of using dask, the internal library is set to 'da'.
+        The chunksize parameter is popped from the argument list, as well as the locking-parameter and the
+        rechunk callback function. Also chunking-related variables are initialized.
+        """
         self.lib = da
-        self.field_chunksize = kwargs.pop('field_chunksize', 'auto')
+        self.chunksize = kwargs.pop('chunksize', 'auto')
         self.lock_file = kwargs.pop('lock_file', True)
         self.chunk_mapping = None
         self.rechunk_callback_fields = kwargs.pop('rechunk_callback_fields', None)
         self.chunking_finalized = False
-        if "chunkdims_name_map" in kwargs.keys() and kwargs["chunkdims_name_map"] is not None and isinstance(kwargs["chunkdims_name_map"], dict):
-            for key, dim_name_arr in kwargs["chunkdims_name_map"].items():
-                for value in dim_name_arr:
-                    if value not in self._name_maps[key]:
-                        self._name_maps[key].append(value)
+        self.autochunkingfailed = False
         super(DaskFileBuffer, self).__init__(*args, **kwargs)
 
     def __enter__(self):
-        if self.field_chunksize not in [False, None, 'auto'] and type(self.field_chunksize) not in [list, tuple, dict]:
-            raise AttributeError("'field_chunksize' is of wrong type. Parameter is expected to be a list, tuple or dict per data dimension, or be False, None or 'auto'.")
-        if isinstance(self.field_chunksize, list):
-            self.field_chunksize = tuple(self.field_chunksize)
+        """
+        This function enters the physical file (equivalent to a 'with open(...)' statement) and returns a file object.
+        In Dask, with dynamic loading, this is the point where we have access to the header-information of the file.
+        Hence, this function initializes the dynamic loading by parsing the chunksize-argument and maps the requested
+        chunksizes onto the variables found in the file. For auto-chunking, educated guesses are made (e.g. with the
+        dask configuration file in the background) to determine the ideal chunk sizes. This is also the point
+        where - due to the chunking, the file is 'locked', meaning that it cannot be simultaneously accessed by
+        another process. This is significant in a cluster setup.
+        """
+        if self.chunksize not in [False, None, 'auto'] and type(self.chunksize) is not dict:
+            raise AttributeError("'chunksize' is of wrong type. Parameter is expected to be a list, tuple or dict per data dimension, or be False, None or 'auto'.")
```

Comment:
Are list and tuple still allowed? Would it not be even simpler to only support dicts?

Reply:
oh, tuples and lists are not supported anymore.